tests/test_switch.py

```diff
@@ -154,3 +164,23 @@ async def test_switch_unique_id(hass: HomeAssistant) -> None:
     assert (
         registry_entry.unique_id == f"{TEST_CONFIG_ENTRY_ID}:switch:front_door_detect"
     )
+
+
+async def test_switch_improve_contrast_can_be_enabled(hass: HomeAssistant) -> None:
```

Comment:
Even better would be to use `@pytest.mark.parametrize` here so that this test gets run for all members of `DISABLED_SWITCH_ENTITY_IDS` (so that when we add more switches there this test will "just work").

Take a look at the sensor test for inspiration: https://github.com/blakeblackshear/frigate-hass-integration/blob/master/tests/test_sensor.py#L92

Reply:
Great idea. So, something like this? Not sure if I've got the syntax right.

```
@pytest.mark.parametrize("disabled_switch_name", DISABLED_SWITCH_ENTITY_IDS)
async def test_disabled_switch_can_be_enabled(
    disabled_switch_name: Any, hass: HomeAssistant
) -> None:
    """Verify disabled switches can be enabled."""
    await setup_mock_frigate_config_entry(hass)
    entity_registry = er.async_get(hass)

    # Test original entity is disabled as expected
    entry = entity_registry.async_get(disabled_switch_name)
    assert entry
    assert entry.disabled
    assert entry.disabled_by == er.DISABLED_INTEGRATION
    entity_state = hass.states.get(disabled_switch_name)
    assert not entity_state

    # Update and test that entity is now enabled
    updated_entry = entity_registry.async_update_entity(
        disabled_switch_name, disabled_by=None
    )
    assert not updated_entry.disabled
```

Really appreciate your patience as I learn this!

Reply:
Yeah, exactly.

Minor:

* I'd call `disabled_switch_name` as `disabled_entity_id`.
* I think you should be able to type annotate it with `str` instead of `Any`